frontend/models/product/server.ts

```diff
@@ -45,13 +52,14 @@ export async function findProduct({
          ),
       ]);
 
+   const urlRedirect = shopifyQueryResponse.urlRedirects.edges[0]?.node?.target;
    const product = await getProduct({
       shopifyProduct: shopifyQueryResponse.product,
       strapiProduct: strapiQueryResponse.products?.data[0],
       iFixitProduct: iFixitQueryResponse,
    });
    if (product == null) {
-      return null;
+      return urlRedirect ? { target: urlRedirect } : null;
```

Comment:
I appreciate this will only return the redirect if a product isn't found.

Reply:
That's true. We could add a `__typename` key or something.

Reply:
cc4c4b084b1212bb2db62374eda606dcd8b9f060